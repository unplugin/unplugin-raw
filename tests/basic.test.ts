import path from 'node:path'
import { build } from 'esbuild'
import { rollup } from 'rollup'
import { expect, test } from 'vitest'
import Raw from '../src'

const resolveDir = path.resolve(__dirname, 'fixtures')

test('esbuild', async () => {
  const contents = `
  import text from "./ts.ts?raw"
  import text2 from "./js.js?raw"
  import text3 from "./jsx.jsx?raw"
  import text4 from "./with.js" with { type: "text" }
  import bytes1 from "./abc.txt" with { type: "bytes" }
  console.log(text, text2, text3, text4, bytes1)
  `

  const result = await build({
    stdin: {
      contents,
      resolveDir,
    },
    write: false,
    bundle: true,
    format: 'esm',
    plugins: [Raw.esbuild()],
  })
  expect(result.outputFiles[0].text).matchSnapshot()
})

test('rollup', async () => {
  const contents = `
  import text from "./ts.ts?raw"
  import text2 from "./js.js?raw"
  import text3 from "./jsx.jsx?raw"
  import text4 from "./with.js" with { type: "text" }
  console.log(text, text2, text3, text4)
  `
  const entryFile = path.resolve(resolveDir, 'main.js')
  const bundle = await rollup({
    input: [entryFile],
    plugins: [
      Raw.rollup(),
      {
        name: 'entry',
        resolveId(id) {
          if (id === entryFile) {
            return entryFile
          }
        },
        load(id) {
          if (id === entryFile) {
            return contents
          }
        },
      },
    ],
  })
  const result = await bundle.generate({ format: 'esm' })
  expect(result.output[0].code).matchSnapshot()
})
