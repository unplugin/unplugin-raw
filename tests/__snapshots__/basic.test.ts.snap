// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild 1`] = `
"var __toBinary = /* @__PURE__ */ (() => {
  var table = new Uint8Array(128);
  for (var i = 0; i < 64; i++) table[i < 26 ? i + 65 : i < 52 ? i + 71 : i < 62 ? i - 4 : i * 4 - 205] = i;
  return (base64) => {
    var n = base64.length, bytes = new Uint8Array((n - (base64[n - 1] == "=") - (base64[n - 2] == "=")) * 3 / 4 | 0);
    for (var i2 = 0, j = 0; i2 < n; ) {
      var c0 = table[base64.charCodeAt(i2++)], c1 = table[base64.charCodeAt(i2++)];
      var c2 = table[base64.charCodeAt(i2++)], c3 = table[base64.charCodeAt(i2++)];
      bytes[j++] = c0 << 2 | c1 >> 4;
      bytes[j++] = c1 << 4 | c2 >> 2;
      bytes[j++] = c2 << 6 | c3;
    }
    return bytes;
  };
})();

// tests/fixtures/ts.ts?raw
var ts_default = 'export const msg = "hello ts";\\n';

// tests/fixtures/js.js?raw
var js_default = 'export const msg = "hello js";\\n';

// tests/fixtures/jsx.jsx?raw
var jsx_default = 'export const msg = /* @__PURE__ */ React.createElement("div", null, "hello jsx");\\n';

// tests/fixtures/with.js
var with_default = 'export const msg = "hello with";\\n';

// tests/fixtures/abc.txt
var abc_default = __toBinary("YWJjCg==");

// <stdin>
console.log(ts_default, js_default, jsx_default, with_default, abc_default);
"
`;

exports[`rolldown 1`] = `
"//#region tests/fixtures/ts.ts?raw
var ts_default = "export const msg = \\"hello ts\\";\\n";

//#endregion
//#region tests/fixtures/js.js?raw
var js_default = "export const msg = \\"hello js\\";\\n";

//#endregion
//#region tests/fixtures/jsx.jsx?raw
var jsx_default = "export const msg = /* @__PURE__ */ React.createElement(\\"div\\", null, \\"hello jsx\\");\\n";

//#endregion
//#region tests/fixtures/main.js
console.log(ts_default, js_default, jsx_default);

//#endregion"
`;

exports[`rollup 1`] = `
"var text = "export const msg = \\"hello ts\\";\\n";

var text2 = "export const msg = \\"hello js\\";\\n";

var text3 = "export const msg = /* @__PURE__ */ React.createElement(\\"div\\", null, \\"hello jsx\\");\\n";

var text4 = "export const msg = \\"hello with\\";\\n";

console.log(text, text2, text3, text4);
"
`;

exports[`vite 1`] = `
"const text = 'export const msg = "hello ts";\\n';
const text2 = 'export const msg = "hello js";\\n';
const text3 = 'export const msg = /* @__PURE__ */ React.createElement("div", null, "hello jsx");\\n';
const text4 = 'export const msg = "hello with";\\n';
console.log(text, text2, text3, text4);
"
`;
